src/Models/Links/AbstractCompoundsLinks.php

```diff
@@ -109,34 +99,67 @@ public function destroy(): bool
         $req = $this->Db->prepare($sql);
         $req->bindParam(':compound_id', $this->id, PDO::PARAM_INT);
         $req->bindParam(':entity_id', $this->Entity->id, PDO::PARAM_INT);
-        return $this->Db->execute($req);
-    }
-
-    public function destroyAll(): bool
-    {
-        $sql = 'DELETE FROM ' . $this->getTable() . ' WHERE entity_id = :entity_id';
-        $req = $this->Db->prepare($sql);
-        $req->bindParam(':entity_id', $this->Entity->id, PDO::PARAM_INT);
-        return $this->Db->execute($req);
+        $res = $this->Db->execute($req);
+        if ($res && $req->rowCount() > 0) {
+            $this->Entity->touch();
+            $this->createChangelog(true);
+        }
+        return $res;
     }
 
-    /**
-     * Add a link to an entity
-     * Links to Items are possible from all entities
-     * Links to Experiments are only allowed from other Experiments and Items
-     */
+    // Add a compound to an entity
     public function create(): int
     {
+        $this->Entity->canOrExplode('write');
         // use IGNORE to avoid failure due to a key constraint violations
         $sql = 'INSERT IGNORE INTO ' . $this->getTable() . ' (compound_id, entity_id) VALUES(:link_id, :item_id)';
         $req = $this->Db->prepare($sql);
         $req->bindParam(':item_id', $this->Entity->id, PDO::PARAM_INT);
         $req->bindParam(':link_id', $this->id, PDO::PARAM_INT);
 
-        $this->Db->execute($req);
+        $res = $this->Db->execute($req);
+        if ($res && $req->rowCount() > 0) {
+            $this->Entity->touch();
+            $this->createChangelog();
+        }
 
         return $this->id;
     }
 
     abstract protected function getTable(): string;
+
+    protected function getTemplateTable(): string
+    {
+        if ($this->Entity instanceof Items || $this->Entity instanceof ItemsTypes) {
+            return 'compounds2items_types';
+        }
+        return 'compounds2experiments_templates';
+    }
+
+    private function createChangelog(bool $isDestroy = false): void
```

Comment:
use Action::Add or Action::Destroy instead of a bool. This also makes it more clear on the call sites what this arg is about.